### ISSUE TITLE
Research & verify existing projects

### ISSUE THREAD

**Issue by zerospiel** [Mirantis/hmc#600]
- velero
- stashed/stash (highly likely no due to community/enterprise separation)

check docs/capabilities, what's available, will it cover our needs

**Comment by zerospiel** [comment]
TLDR: [Velero](https://velero.io/) > [Stash](https://stash.run/), the former is easier to use (from a user perspective) and has the stable version

- Stash is still in development and that might cause potential problems (breaking changes) either with the packages or its own API
- Stash is a MUCH more file-oriented solution, meaning it's hard to spin and run it from scratch, it requires too much preparation with different custom resource objects just to create a simple backup
- because of the latter, Stash has more granular backup settings compared to Velero, which we do not need at the moment
- both of the solutions have similar features, though Velero has no support for backing databases up but Stash only has it in the Enterprise Edition
- Stash has 2 models: community and enterprise, so in theory this might be a small problem, though none of the enterprise's features are not required for our needs
- ❗ Velero relies mostly on its CLI usage so it's hard to say whether it would be more beneficial from the code perspective compared to Stash. From the API perspective, both have the same amount of CRDs (Stash requires more of them to establish an initial backup for example)

@DinaBelova @Kshatrix let me know WDYT about this. I have expressed my opinion, that as a user I prefer Velero but I haven't delved into the code of both products to understand which might be easier to utilize, however, Velero looks simpler to me CRD-wise

I am closing the issue at the moment, feel free to reopen if you have questions/want me to delve into the code

**Comment by moshiurH** [comment]
@zerospiel @DinaBelova @Kshatrix 
I want to raise a concern about Velero. Their [documentation](https://velero.io/docs/v1.15/supported-providers/) says the following:

![Image](https://github.com/user-attachments/assets/c55431f8-f9a4-4aa6-aaf4-dd5c2df47306)

I believe the goal for k0rdent is to support AWS, Azure, vSphere and OpenStack.

Given Velero does not support VMware vsphere's object store (I think it's called vSphere Object Storage)? Do we need to perhaps document the limitation for [backup&restore](https://docs.k0rdent.io/head/admin-backup/) for VMware vsphere? Would this be a concern for customers? Should we recommend customers on VMware vsphere to use an object storage that is supported by Velero (like AWS S3 bucket)?

**Comment by zerospiel** [comment]
@moshiurH thanks for the suggestion, i stated it explicitly https://github.com/k0rdent/docs/pull/107 that it is required from a plugin to support object store feature in order to put backups in an object storage
